images/build/go-runner/Dockerfile

```diff
@@ -48,13 +48,16 @@ ENV GOARCH=${ARCH}
 
 RUN go env
 
-RUN go build -ldflags '-s -w -buildid= -extldflags "-static"' \
-    -o go-runner ${package}
+RUN GOBIN=`pwd` go install -ldflags '-s -w -buildid= -extldflags "-static"' \
```

Comment:
`make REGISTRY=localhost:5000 push` got passed this for me for GOARCH=amd64, but failed for GOARCH=arm while downloading the package:
```
#10 [builder 4/5] RUN go env
#10 1.174 GO111MODULE=""
#10 1.176 GOARCH="arm"
...
#11 [builder 5/5] RUN GOBIN=`pwd` go install -ldflags '-s -w -buildid= -extl...
#11 0.385 go: downloading k8s.io/component-base v0.0.0-20211124233101-2083abd532f6
#11 57.68 create zip: zip: checksum error
#11 ERROR: process "/bin/sh -c GOBIN=`pwd` go install -ldflags '-s -w -buildid= -extldflags \"-static\"'     ${package}" did not complete successfully: exit code: 1
------
 > [builder 5/5] RUN GOBIN=`pwd` go install -ldflags '-s -w -buildid= -extldflags "-static"'     k8s.io/component-base/logs/kube-log-runner:
------
failed to solve: rpc error: code = Unknown desc = process "/bin/sh -c GOBIN=`pwd` go install -ldflags '-s -w -buildid= -extldflags \"-static\"'     ${package}" did not complete successfully: exit code: 1
make: *** [/nvme/gopath/src/k8s.io/release/images/build/go-runner/../../Makefile.common-image:41: container] Error 1
```

I'm probably missing some local support for ARM.


```

Reply:
Hmm, the same problem occurs in `pull-release-image-go-runner` (https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/release/2333/pull-release-image-go-runner/1463874201663311872).

Any suggestions?

Reply:
I added one commit with a workaround (download source in advance, build without module support). That helps a bit, but now it fails for s390x:

```
Step #1: #11 [builder 5/5] RUN GO111MODULE=off go build -o kube-log-runner -ldflags '-s -w -buildid= -extldflags "-static"'     ./vendor/k8s.io/component-base/logs/kube-log-runner
Step #1: #11 2.283 # _/workspace/vendor/k8s.io/component-base/logs/kube-log-runner
Step #1: #11 2.283 fatal error: unexpected signal during runtime execution
Step #1: #11 2.283 fatal error: unexpected signal during runtime execution
Step #1: #11 2.283 unexpected fault address 0xffffffffbe0cb000
Step #1: #11 2.283 fatal error: fault
Step #1: #11 2.283 fatal error: unexpected signal during runtime execution
Step #1: #11 2.283 panic during panic
Step #1: #11 2.283 [signal SIGSEGV: segmentation violation code=0x1 addr=0x97e0091c000 pc=0x5986a]
```

Reply:
The same failure also occurs without this PR:
- https://github.com/kubernetes/release/pull/2336
- https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/release/2336/pull-release-image-go-runner/1465278525853405184/build-log.txt

Reply:
@BenTheElder : any idea? ^^^

Looks like there's a general problem with building for s390x.

Reply:
/retest

Perhaps something changed...